MovieApp/MovieApp/NewsFeedTableViewController.m

```diff
@@ -0,0 +1,121 @@
+#import "FeedTableViewCell.h"
+#import "NewFeedsItem.h"
+#import "FeedDownloader.h"
+
+@interface NewsFeedTableViewController (){
+    NSMutableArray *news;
+}
+
+@end
+
+@implementation NewsFeedTableViewController
+
++(NSURL *)getFeedsSource
+{
+    return [NSURL URLWithString:@"http://www.boxofficemojo.com/data/rss.php?file=topstories.xml"];
+}
+
+-(void)updateViewWithNewData:(NSMutableArray *)feedItemsArray{
+    news=feedItemsArray;
+    
+    if(![news count])
+    {
+        if(!news)
+        {
+            news = [[NSMutableArray alloc] init];
+        }
+        [news addObject:[[NewFeedsItem alloc] initWithHeadline:@"nema" andWithText:@"rezultata" andWithWebPage:@"sry"]];
+    }
+    [self.tableView reloadData];
+    
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    /*
+     //hardcoded items
+     news = [[NSMutableArray alloc] init];
+    
+    NewFeedsItem *item = [[NewFeedsItem alloc] initWithHeadline:@"Testni naslov" andWithText:@"Lorem ipsum dolor sit er elit lamet, consectetaur cillium adipisicing pecu, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Nam liber te conscient to factor tum poen legum odioque civiuda." andWithWebPage:@"www.klix.ba"];
+    
+    for(int i=0;i<5;i++){
+        [news addObject:item];
+    }
+     */
+   
+     [self.tableView registerNib:[UINib nibWithNibName:@"FeedTableViewCell" bundle:nil] forCellReuseIdentifier:[FeedTableViewCell cellIdentifier]];
+    
+    FeedDownloader *downloader = [[FeedDownloader alloc]init];
+    
+    @try {
+        [downloader downloadNewsFromFeed:[NewsFeedTableViewController getFeedsSource] andReturnTo:self];
+        
+    } @catch (NSException *exception) {
+        [news removeAllObjects];
+        [news addObject:[[NewFeedsItem alloc] initWithHeadline:@"Error" andWithText:[exception description]
+                                                andWithWebPage:@""]];
+    } 
+    
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return [news count];
+}
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    FeedTableViewCell *feedCell = [tableView dequeueReusableCellWithIdentifier:[FeedTableViewCell cellIdentifier]];
+    
+    if(!feedCell)
+    {
+       
+        feedCell = [tableView dequeueReusableCellWithIdentifier:@"feedCell"];
+
+    }
```

Comment:
You can remove 

```
if(!feedCell)
 {
        feedCell = [tableView dequeueReusableCellWithIdentifier:@"feedCell"];
 }
```

since it will never be called. you should always dequeue specific cell, that is registered in `viewDidLoad`.